camelot/view/workspace.py

```diff
@@ -60,15 +60,21 @@ class DesktopWorkspace(QtWidgets.QWidget):
     .. attribute:: background
 
     The widget class to be used as the view for the uncloseable 'Start' tab.
+
+    :param app_admin: the application admin object for this application
+    :param parent: a :class:`QtWidgets.QWidget` object or :class:`None`
+    :param main_window: a :class:`camelot.view.mainwindow.MainWindow` object or :class:`None`
+
     """
 
     view_activated_signal = QtCore.qt_signal(QtWidgets.QWidget)
 
-    def __init__(self, app_admin, parent):
+    def __init__(self, app_admin, parent, main_window):
         super(DesktopWorkspace, self).__init__(parent)
         self.gui_context = ApplicationActionGuiContext()
         self.gui_context.admin = app_admin
         self.gui_context.workspace = self
+        self.main_window = main_window
```

Comment:
het is best om zo weinig mogelijk python references te hebben naar Qt objects

dwz, hier kan je beter de main_window gebruiken als parent voor de DesktopWorkspace, en als je de mainwindow nodig hebt, dan neem je parent()

Reply:
ah sorry, ik heb dat hier verkeerd gelezen main_window is geen QObject, maar onze 'builder'